### PR TITLE
feat: add react tooltip component

### PR DESCRIPTION
**Pull request by hamza14khan**
- Addition of react tooltip component.
- Unit tests.
- Documentation update.
- Storybook update.
- 4 variants, top, bottom, left and right.

Storybook testing:
Screenshot 2024-12-11 at 14 35 06

### CODE REVIEW

**Review by pregno — APPROVED** [review]

**Review by pregno — CHANGES_REQUESTED** [review]
LGTM, just added a comment for the doc

File: apps/docs/content/3-components/2-library/tooltip.mdx [hunk 1]
```diff
@@ -9,22 +9,70 @@ libraries:
...
+    status: 'beta'
+    link: ?path=/docs/application-tooltip--docs'
 ---
 
 # Tooltip
 
-## Variants
+## Default
+
+<Tooltip label="Default Tooltip" position="top">
```
[the remark this answers is not in the document]
Comment (hamza14khan) [reply apps/docs/content/3-components/2-library/tooltip.mdx:20]: Good catch!

**Review by BogdanDanilescu — APPROVED** [review]
LGTM

File: packages/react/ds/src/tooltip/tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  return (
+    <span
+      className="gi-tooltip-wrapper"
+      onMouseEnter={showTooltip}
+      onMouseLeave={hideTooltip}
+    >
+      {props.children}
+      {isVisible && (
+        <span className={`gi-tooltip gi-tooltip-${props.position}`}>
```
Comment (ramonmulia) [inline packages/react/ds/src/tooltip/tooltip.tsx:24]: us tw variants library for that or maybe you should put in safe list?

File: apps/docs/content/3-components/2-library/tooltip.mdx [hunk 3]
```diff
@@ -9,22 +9,72 @@ libraries:
...
+    ```
+
+  </TabPanel>
+</Tabs>
+
 ### Best Practices
 
+- Keep the tooltip content concise and relevant. Tooltips are meant for brief contextual information.
+- Avoid overloading tooltips with large amounts of text or complex HTML structures.
+- Ensure the tooltip is accessible by providing appropriate ARIA attributes (`aria-label`, `aria-describedby`).
```
Comment (ramonmulia) [inline apps/docs/content/3-components/2-library/tooltip.mdx:62]: I didn't see the aria attributes on the component, not sure how the developper will use it.

Comment (hamza14khan) [reply apps/docs/content/3-components/2-library/tooltip.mdx:62]: Good catch!

**Review by ramonmulia — APPROVED** [review]
LGTM

**Review by pregno — APPROVED** [review]

**Pull request merged**